frida-gum/src/module_map.rs

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright © 2020-2021 Keegan Saunders
+ * Copyright © 2021 S Rubenstein
+ *
+ * Licence: wxWindows Library Licence, Version 3.1
+ */
+
+use frida_gum_sys as gum_sys;
+use std::{ffi::CStr, os::raw::c_void, path::Path};
+
+use crate::MemoryRange;
+
+pub struct ModuleDetails {
+    module_details: *const gum_sys::GumModuleDetails,
+}
+
+/// Represents a loaded module
+impl ModuleDetails {
+    pub(crate) fn from_raw(module_details: *const gum_sys::_GumModuleDetails) -> Self {
+        Self { module_details }
+    }
+
+    /// Get a new [`ModuleDetails`] instance for the module matching the given name. The name may
+    /// be a full path, in which case the matching module must have the same full path, or a
+    /// file name, in which case only the file name portion of the module must match.
+    pub fn with_name(name: String) -> Option<Self> {
+        struct SaveModuleDetailsByNameContext {
+            name: String,
+            details: *mut gum_sys::GumModuleDetails,
+        }
+
+        unsafe extern "C" fn save_module_details_by_name(
```

Comment:
Please extract this outside of `ModuleDetails` (put it at the top-level)

Reply:
Gotcha. Thought it was cleaner to have it scoped.